packages/autocomplete-plugin-tags/src/createTags.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseItem } from '@algolia/autocomplete-core';
+import { createRef } from '@algolia/autocomplete-shared';
+
+import { CreateTagsPluginParams } from './createTagsPlugin';
+import { BaseTag, Tag } from './types';
+
+export type OnTagsChangeParams<TTag> = {
+  prevTags: Array<Tag<TTag>>;
+  tags: Array<Tag<TTag>>;
+};
+
+type OnTagsChangeListener<TTag> = (params: OnTagsChangeParams<TTag>) => void;
+
+type CreateTagsParams<TTag, TItem extends BaseItem> = Pick<
+  CreateTagsPluginParams<TTag, TItem>,
+  'initialTags'
+>;
+
+export function createTags<TTag, TItem extends BaseItem>({
+  initialTags = [],
+}: CreateTagsParams<TTag, TItem>) {
+  const tagsRef = createRef(toTags(initialTags));
+  const onChangeListeners: Array<OnTagsChangeListener<TTag>> = [];
+
+  function toTags<TTag>(baseTags: Array<BaseTag<TTag>>): Array<Tag<TTag>> {
```

Comment:
How about `mapToTags`? We usually start function names with verbs.

Reply:
Addressed in 04e3eee.